src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,189 @@
+            return false;
+        }
+        Node<T> prev;
+        prev = last;
+        for (T elementAddArray : addArray) {
+            Node<T> inputNode = new Node<>(elementAddArray, null, prev);
+            if (prev == null) {
+                first = inputNode;
+            } else {
+                prev.next = inputNode;
+            }
+            prev = inputNode;
+        }
+        size += lengthOfArray;
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldNode = node(index);
+        T oldValue = oldNode.element;
+        oldNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T t) {
+        if (t == null) {
+            for (Node<T> i = first; i != null; i = i.next) {
+                if (i.element == null) {
+                    unlink(i);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> i = first; i != null; i = i.next) {
+                if (i.element.equals(t)) {
+                    unlink(i);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkFirst(T e) {
+        final MyLinkedList.Node<T> f = first;
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(e, f, null);
+        first = newNode;
+        if (f == null) {
+            last = newNode;
+        } else {
+            f.prev = newNode;
+        }
+        size++;
     }
+
+    void linkLast(T e) {
+        final Node<T> l = last;
+        final Node<T> newNode = new MyLinkedList.Node<>(e, null, l);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+    }
+
+    public void checkIndex(int index) {
+        if (index > size - 1 || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+    }
+
+    public void linkBefore(T element, Node<T> movingElement) {
+        Node<T> currentPrev = movingElement.prev;
+        Node<T> inputNode = new Node<>(element, movingElement, currentPrev);
+        movingElement.prev = inputNode;
+        if (currentPrev == null) {
+            first = inputNode;
+        } else {
+            currentPrev.next = inputNode;
+        }
+        size++;
+    }
+
+    public Node<T> node(int index) {
```

Comment:
```suggestion
    public Node<T> findNode(int index) {
```
what's wrong with the access modifiers for all the helper methods?

Reply:
Fixed.